test/src/e2e_vm_tests/test_programs/should_pass/stdlib/vec/src/main.sw

```diff
@@ -225,6 +225,64 @@ fn test_vector_new_u8() {
     }
     assert(vector.len() == 0);
     assert(vector.capacity() == 4);
+
+    
+    // Test for `set`
+    vector.clear();
+    vector.push(number0);
+    vector.push(number1);
+    vector.push(number2);
+    assert(vector.len() == 3);
+    assert(vector.capacity() == 4);
+
+    // Set at first
+    vector.set(0, number3);
+
+    assert(vector.len() == 3);
+    assert(vector.capacity() == 4);
+    match vector.get(0) {
+        Option::Some(val) => assert(val == number3), Option::None => revert(0), 
+    }
+
+    match vector.get(1) {
+        Option::Some(val) => assert(val == number1), Option::None => revert(0), 
+    }
+
+    match vector.get(2) {
+        Option::Some(val) => assert(val == number2), Option::None => revert(0), 
+    }
```

Comment:
The first step to performing a test is proving the initial conditions are what they are expected to be.
This means that the workflow is as follows:

1. Set up the conditions
2. Prove that the subject of the test has been correctly set in the set up
3. Perform computation
4. Prove that the computation has changed the originally set subject and that it has changed to the expected result

The code in this section, and subsequent sections, sets up the conditions and performs computation without proving the original conditions. 
This could lead to unexpected results depending on the coverage of tests.
For example, if you imagine that the rest of the test suite does not exist and `push` has been changed then it's possible that `push` ignores whatever the first argument is passed in and sets the value of 3 (as does the computation in the new function) and then it works as expected.

If the initial conditions are proven then such a change to `push` would be caught and the test would not falsely pass.

Reply:
Alright, so i'll incorporate additional `assert` checks after using `push` for the set up